builtin/providers/google/resource_compute_router.go

```diff
@@ -0,0 +1,251 @@
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"bgp": &schema.Schema{
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"asn": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceComputeRouterCreate(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+
+	routerId := fmt.Sprintf("router/%s/%s", region, name)
+	mutexKV.Lock(routerId)
+	defer mutexKV.Unlock(routerId)
+
+	network, err := getNetworkLink(d, config, "network")
+	if err != nil {
+		return err
+	}
+	routersService := compute.NewRoutersService(config.clientCompute)
+
+	router := &compute.Router{
+		Name:    name,
+		Network: network,
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		router.Description = v.(string)
+	}
+
+	if _, ok := d.GetOk("bgp"); ok {
+		prefix := "bgp.0"
+		if v, ok := d.GetOk(prefix + ".asn"); ok {
+			asn := v.(int)
+			bgp := &compute.RouterBgp{
+				Asn: int64(asn),
+			}
+			router.Bgp = bgp
+		}
+	}
+
+	op, err := routersService.Insert(project, region, router).Do()
+	if err != nil {
+		return fmt.Errorf("Error Inserting Router %s into network %s: %s", name, network, err)
+	}
+
+	err = computeOperationWaitRegion(config, op, project, region, "Inserting Router")
+	if err != nil {
+		return fmt.Errorf("Error Waiting to Insert Router %s into network %s: %s", name, network, err)
+	}
+
+	return resourceComputeRouterRead(d, meta)
+}
+
+func resourceComputeRouterRead(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	routersService := compute.NewRoutersService(config.clientCompute)
+	router, err := routersService.Get(project, region, name).Do()
+
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
+			log.Printf("[WARN] Removing Router %q because it's gone", d.Get("name").(string))
+			d.SetId("")
+
+			return nil
+		}
+
+		return fmt.Errorf("Error Reading Router %s: %s", name, err)
+	}
+
+	d.Set("self_link", router.SelfLink)
+
+	// if we don't have a network (when importing), set it to the URI returned from the server
```

Comment:
When the user specifies the network just by name, and we use getNetworkLink to get the URL to be passed to the API, setting it back on read will cause a delta that triggers router recreation. Any way to avoid that?